### PR TITLE
Feat/59/sent endpoint question

### PR DESCRIPTION
**Pull request by matt-newhall**

### CODE REVIEW

File: src/app/pages/api/openai.ts [hunk 1]
```diff
@@ -1,52 +1,36 @@
...
-    }),
-  });
-
-  const data = await response.json();
-  return data as OpenaiApiType;
-};
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
```
Comment (will249) [inline src/app/pages/api/openai.ts:31]: nice

File: src/app/utils/extractKnowledge/extractKnowledge.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { SupabaseResponse } from '@/types/SupabaseResponse';
+
+const extractKnowledge = (context: SupabaseResponse[]): string[] => {
+  return context.reduce((acc: string[], knowledge) => {
+    const inputText = knowledge.input_text;
+
+    acc.push(inputText);
+
+    return acc;
```
Comment (will249) [inline src/app/utils/extractKnowledge/extractKnowledge.ts:9]: what's acc?

File: src/app/utils/handleApiRequest/createEmbedding/createEmbedding.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import dotenv from 'dotenv';
 import * as path from 'path';
 import { OpenaiEmbeddingType } from '@/app/types/OpenaiApiType';
-dotenv.config({ path: path.resolve(__dirname, '../../../..', '.env.local') });
+dotenv.config({
+  path: path.resolve(__dirname, '../../../../..', '.env.local'),
+});
```
Comment (will249) [inline src/app/utils/handleApiRequest/createEmbedding/createEmbedding.ts:6]: can this be removed now we're on the main run process?

File: src/app/utils/react-query-hooks.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ export const useItems = (initialData: Array<Table<'items'>>) => {
```
Comment (will249) [inline src/app/utils/react-query-hooks.ts:25]: I think these helper files can be deleted, they were inherited. Sorry, looks like you've had to mess around loads with types here!

File: src/utils/createEmbedding.ts [hunk 5]
```diff
@@ -34,14 +41,18 @@ export const createEmbeddings = async (
...
   const outputArray: EmbeddingsArray = inputChunks.map((inputChunk, index) => {
-    const embedding = embeddings.filter((val) => {
-      return val.index === index;
-    })[0];
-    return {
-      input_url: inputChunk.url,
-      input_text: inputChunk.input_text,
-      vector: JSON.stringify(embedding.embedding),
-    };
+    if (IsCreateEmbeddingResponseDataInnerArray(embeddings)) {
```
Comment (will249) [inline src/utils/createEmbedding.ts:44]: nice!

File: src/app/utils/extractKnowledge/extractKnowledge.ts (hunk 2 above)
Comment (matt-newhall) [reply src/app/utils/extractKnowledge/extractKnowledge.ts:9]: i usually use `acc` in `reduce` functions that output an array - means `accumulated` :)

File: src/app/utils/react-query-hooks.ts (hunk 4 above)
Comment (matt-newhall) [reply src/app/utils/react-query-hooks.ts:25]: ah fair! will do

**Review by will249 — APPROVED** [review]

**Pull request merged**